src/Analysis/Ast/Impl/Types/PythonUnionType.cs

```diff
@@ -65,10 +65,26 @@ internal sealed class PythonUnionType : IPythonUnionType {
         public IMember CreateInstance(string typeName, LocationInfo location, IArgumentSet args)
             => new PythonUnion(this, location);
 
-        public IMember Call(IPythonInstance instance, string memberName, IArgumentSet args)
-            => DeclaringModule.Interpreter.UnknownType;
-        public IMember Index(IPythonInstance instance, object index)
-            => DeclaringModule.Interpreter.UnknownType;
+        public IMember Call(IPythonInstance instance, string memberName, IArgumentSet args) {
+            lock (_lock) {
+                // Check if any types support calls
+                var result = _types
+                    .Select(t => t.Call(instance, memberName, args))
```

Comment:
A bit scary to call `Call` under the lock. Can we move it out?

Reply:
Sure although it is a simulated call in the same analysis.